src/Parser/Daedalus.g4

```diff
@@ -102,14 +103,16 @@ value
     | NoFunc #noFuncLiteralValue
     | funcCall #funcCallValue
     | reference #referenceValue
+    | nameNode #anyIdentifierValue
     ;
     
-referenceAtom: Identifier ( '[' arrayIndex ']')?;
+referenceAtom: nameNode ( '[' arrayIndex ']')?;
 reference: referenceAtom ( '.' referenceAtom )?;
 
 typeReference:  ( Identifier | Void | Int | Float | String | Func | Instance);
+anyIdentifier:  ( Void | Int | Float | String | Func | Instance | Class | Prototype | Null | Identifier);
```

Comment:
What's the purpose of this? It looks like very messy.

Reply:
the `anyIdentifier` is neccessery, because of the lexer consuming the tokens that would make up field-names or method parameters. (i did not currently look around much if there is another way around that)
also there are classes/instances in LeGo for example, that contain fields named `class` or parameters named `instance` or `class`

think of methods like `func testFunc(var c_npc instance) {}`